design_docs/timekeeping_reorg.md

```diff
@@ -0,0 +1,182 @@
+<h1> Reorganize Timekeeping <br>
+Xylar Asay-Davis <br>
+date: 2017/02/06 <br>
+</h1>
+<h2> Summary </h2>
+Currently, a the `Date` class is used to parse a date object from a date string
+(e.g. '0001-01-01_00:00:00') taken from MPAS namelists, streams files or time
+variables (e.g. `xtime`).  However, this class assumes a 365-day calendar and
+cannot easily be adapted to the Gregorian calendar also supported by MPAS
+components.  Furthermore, existing routines exist to handle most of the capabilites
+of the `Date` class.  The proposed reorganization would eliminate the `Date` class
+in favor of a numer of helper functions that can be used to convert between various
+date formats: date strings, days since a reference date, `datetime.datetime` objects
+and `relativedelta` objects (see below).  The success of this reorganization will be
+demonstrated when the existing analysis can be performed successfully with the new
+utility functions with both the `'gregorian_noleap'` calendar used by most existing
+ACME and MPAS runs and the `'gregorian'` calendar also supported in MPAS components.
+
+
+<h1> Requirements </h1>
+
+<h2> Requirement: Date string parsing supports both MPAS calendars <br>
+Date last modified: 2017/02/06 <br>
+Contributors: Xylar Asay-Davis
+</h2>
+
+There must be a way to parse dates from MPAS that is aware of the appropriate calendar to
+use, either `'gregorian'` or `'gregorian_noleap'`.
```

Comment:
No, I mean there should be a way to use the namelist option `config_calendar_type` to determine which calendar is correct, and then to parse date strings accordingly.  I will clarify.